### ISSUE TITLE
Question on clusters and doubly robust scores

### ISSUE THREAD

**Issue by hhsievertsen** [grf-labs/grf#1026]
Hi team,

I have atwo questions about clustering and doubly robust scores. It might be better to ask two separate questions, but to some extent they are linked, so here it goes

1. I use the `clusters=` option when I train the forest. Should I be worried if I compute average treatment effects on subsets that do not nest clusters? I am thinking of this sentence in the documentation: _"When clustering is enabled during training, all subsampling procedures operate on entire clusters as opposed to individual examples."_ 
For example: I use `policy_tree(...)` to find the best policy that goes across clusters and then use `average_treatment_effect(cf,subset=policystatus==2)` to compute the ATE of those who are treated by the policy. 

2. Is there a convenient mapping between the doubly robust scores and the AIPW ATEs? For example I initially reported the benefits of various policies by computing the means of the doubly robust scores for treated and untreated subsets (where treated and untreated are based on the optimal policy), but I then changed my mind to report the benefits based on `average_treatment_effect(cf,subset=policystatus==2)`  which gives **slightly** (not more than I would expect, but still..) different results. I then started to think about differences between the two approaches, which also brought me back to the clustering. 

Sorry if the questions are a) trivial and/or b) confusing. Below is an example of what I am doing

```
library(grf)
library(policytree)
library(dplyr)
set.seed(1234)
# Generate data.
n <- 2000
p <- 3
X <- matrix(rnorm(n * p), n, p)
# Cluster 
df_cluster<-data.frame(cluster=floor(runif(n)*50),e=rnorm(n,0,1))%>%
            group_by(cluster)%>%mutate(e=e[1])
# Train a causal forest.
W <- rbinom(n, 1, 0.4 + 0.2 * (X[, 1] > 0))
Y <- pmax(X[, 1], 0) * W + X[, 2] + pmin(X[, 3], 0) + rnorm(n)+df_cluster$e
cf<- causal_forest(X, Y, W)
cf_cluster<- causal_forest(X, Y, W,clusters=df_cluster$cluster)

# ATES
average_treatment_effect(cf)
average_treatment_effect(cf_cluster)


# Optimal policy
dr.scores <- double_robust_scores(cf)
tree <- policy_tree(X, dr.scores, depth = 2)
policystatus<-predict(tree,X)

# Benefit for treated using dr scores
mean(dr.scores[policystatus==2,2]-dr.scores[policystatus==2,1])
# Benefit for treated using average treatment effect
average_treatment_effect(cf,subset=policystatus==2)
```

Thanks a lot for developing both the papers and the packages. And thanks for responding so quickly to questions! Really very helpful!

Hans

**Comment by erikcs** [comment]
Hi @hhsievertsen, 1) sounds reasonable, the only extreme case that would raise an error is when the subset passed to `average_treatment_effect` contains only units from a single cluster: recall that with interpretation that clustering reduces the unit of observation to each cluster, asking for a mean/std.err of only one cluster amounts to asking for a mean/std.err of just a single observation, which isn't well defined.

For 2) you are completely correct, these two are identical: `double_robust_scores` gives DR scores for Y0 and Y1, `average_treatment_effect` computes DR scores for Y1 - Y0. You can get access the to DR scores used for AIPW inside `average_treatment_effect` using the new (grf 2.0+) `get_scores` function, and you can see that `dr.scores[, 2] - dr.scores[, 1] - get_scores(cf) == 0` (~0 within some machine tolerance, which is what you are seeing from the last two lines: numerically they should be the same with some 1e^-10 epsilon?). The only difference between these two is practical at the moment, if you first estimate your policy on a training set, then passing the data to `average_treatment_effect` on a test set wouldn't work as that function currently does not take a test `X` argument.

**Comment by hhsievertsen** [comment]
Hi @erikcs, 
1) great. Thanks. I had not thought about the SE issue!
2) thanks! 

Hans
